File: Week 07/id_533/LeetCode_190.js
// https://leetcode-cn.com/problems/reverse-bits/
// JS对于整数的位运算仅支持32位有符号整数

/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function(n) {
    var ans = 0;
    for (var i = 0; i < 32; i++) {
        ans = (ans << 1) | (n & 1);
        n >>= 1;
    }
    return ans >>> 0;
};

File: Week 07/id_113/LeetCode190.py
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        ans, MASK = 0, 1
        for i in range(32):
            if n&MASK:
                ans |= 1 << (31 - i)
            MASK <<= 1
        return ans